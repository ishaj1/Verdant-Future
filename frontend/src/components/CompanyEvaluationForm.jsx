import { useState } from "react";
import axios from "../api/axios";
import { useNavigate } from "react-router-dom";
import useAuth from "../hooks/useAuth";

function CompanyEvaluationForm() {
  const currentDate = new Date();
  const offset = currentDate.getTimezoneOffset() * 60000; // Get offset in milliseconds
  const currentDateLocal = new Date(currentDate - offset).toISOString();

  const [errors, setErrors] = useState("");

  const navigate = useNavigate();

  const { auth } = useAuth();

  const evaluateCompany = (e) => {
    e.preventDefault();
    const form = e.currentTarget;
    const companySize = form.companySize.value;
    const revenue = form.revenue.value;
    const emission = form.emission.value;
    const electricity = form.electricity.value;
    const naturalGas = form.naturalGas.value;
    const water = form.water.value;
    const waste = form.waste.value;
    const recycled = form.recycled.value;

    const evaluationData = {
      username: auth.username,
      company_size: companySize,
      revenue: revenue,
      date: currentDateLocal,
      emission: emission,
      electricity: electricity,
      natural_gas: naturalGas,
      water: water,
      waste: waste,
      recycled: recycled,
    };

    axios
      .post("/get_evaluated", evaluationData, {
        headers: { "Content-Type": "application/x-www-form-urlencoded" },
      })
      .then((response) => {
        if (response.data.evaluate == true) {
          navigate("/EvaluationSuccess");
        } else {
          setErrors("Error with evaluation process");
        }
      })
      .catch((error) => {
        setErrors("Error submitting evaluation. Please try again.");
      });
  };

  return (
    <div className="EvaluationPage">
      <h2>Company Evaluation Form</h2>
      <p>{errors}</p>
      <form className="EvaluationForm" onSubmit={(e) => evaluateCompany(e)}>
        <div>
          <label htmlFor="companySize">Number of Employees:</label>
          <input type="number" name="companySize" min="1" required />
        </div>

        <div>
          <label htmlFor="revenue">Revenue (USD):</label>
          <input type="number" name="revenue" min="0" required />
        </div>

        <div>
          <label htmlFor="date">Evaluation Date:</label>
          <input
            type="date"
            name="date"
            value={currentDateLocal.split("T")[0]}
            required
            disabled
          />
        </div>

        <div>
          <label htmlFor="emission">Emission (metric ton):</label>
          <input type="number" name="emission" min="0" required />
        </div>

        <div>
          <label htmlFor="electricity">Electricity (kWh):</label>
          <input type="number" name="electricity" min="0" required />
        </div>

        <div>
          <label htmlFor="naturalGas">Natural Gas (cf):</label>
          <input type="number" name="naturalGas" min="0" required />
        </div>

        <div>
          <label htmlFor="water">Water (cubic meter):</label>
          <input type="number" name="water" min="0" required />
        </div>

        <div>
          <label htmlFor="waste">Total waste (kg):</label>
          <input type="number" name="waste" min="0" required />
        </div>

        <div>
          <label htmlFor="recycled">Recycled waste (kg):</label>
          <input type="number" name="recycled" min="0" required />
        </div>

        <button type="submit">Submit</button>
      </form>
    </div>
  );
}

export default CompanyEvaluationForm;
