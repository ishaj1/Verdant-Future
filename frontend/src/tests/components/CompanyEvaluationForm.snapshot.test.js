import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from "react-router-dom";
import CompanyEvaluationForm from '../../components/CompanyEvaluationForm';

describe('CompanyEvaluationForm snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Router><CompanyEvaluationForm /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
