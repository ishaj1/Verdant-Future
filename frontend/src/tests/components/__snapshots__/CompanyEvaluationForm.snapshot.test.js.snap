// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyEvaluationForm snapshot renders correctly 1`] = `
<div
  className="EvaluationPage"
>
  <h2>
    Company Evaluation Form
  </h2>
  <p />
  <form
    className="EvaluationForm"
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="companySize"
      >
        Number of Employees:
      </label>
      <input
        id="companySize"
        min="1"
        name="companySize"
        required={true}
        type="number"
      />
    </div>
    <div>
      <label
        htmlFor="revenue"
      >
        Revenue (USD):
      </label>
      <input
        id="revenue"
        min="0"
        name="revenue"
        required={true}
        type="number"
      />
    </div>
    <div>
      <label
        htmlFor="date"
      >
        Evaluation Date:
      </label>
      <input
        disabled={true}
        id="date"
        name="date"
        required={true}
        type="date"
        value="2024-04-30"
      />
    </div>
    <div>
      <label
        htmlFor="emission"
      >
        Emission (metric ton):
      </label>
      <input
        id="emission"
        min="0"
        name="emission"
        required={true}
        type="number"
      />
    </div>
    <div>
      <label
        htmlFor="electricity"
      >
        Electricity (kWh):
      </label>
      <input
        id="electricity"
        min="0"
        name="electricity"
        required={true}
        type="number"
      />
    </div>
    <div>
      <label
        htmlFor="naturalGas"
      >
        Natural Gas (cf):
      </label>
      <input
        id="naturalGas"
        min="0"
        name="naturalGas"
        required={true}
        type="number"
      />
    </div>
    <div>
      <label
        htmlFor="water"
      >
        Water (cubic meter):
      </label>
      <input
        id="water"
        min="0"
        name="water"
        required={true}
        type="number"
      />
    </div>
    <div>
      <label
        htmlFor="waste"
      >
        Total waste (kg):
      </label>
      <input
        id="waste"
        min="0"
        name="waste"
        required={true}
        type="number"
      />
    </div>
    <div>
      <label
        htmlFor="recycled"
      >
        Recycled waste (kg):
      </label>
      <input
        id="recycled"
        min="0"
        name="recycled"
        required={true}
        type="number"
      />
    </div>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
